hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/defaultfreezestatus/DefaultFreezeStatusCall.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.defaultfreezestatus;
+
+import static com.hedera.hapi.node.base.ResponseCodeEnum.SUCCESS;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.HederaSystemContract.FullResult.successResult;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.defaultfreezestatus.DefaultFreezeStatusTranslator.DEFAULT_FREEZE_STATUS;
+
+import com.hedera.hapi.node.state.token.Token;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.HederaSystemContract;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.AbstractTokenViewCall;
+import com.hedera.node.app.service.contract.impl.hevm.HederaWorldUpdater;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+public class DefaultFreezeStatusCall extends AbstractTokenViewCall {
+    public DefaultFreezeStatusCall(
+            @NonNull final HederaWorldUpdater.Enhancement enhancement, @Nullable final Token token) {
+        super(enhancement, token);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected @NonNull HederaSystemContract.FullResult resultOfViewingToken(@Nullable final Token token) {
```

Comment:
Here I think we may need a new variant of `AbstractTokenViewCall` that instead of reverting [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/AbstractTokenViewCall.java#L45) on a null `token`, does e.g.
```
if (token == null) {
    return gasOnly(viewCallResultWith(INVALID_TOKEN_ID, 0L));
}
```
where there's a method
```
abstract FullResult viewCallResultWith(ResponseCodeEnum status);
```
that subclasses need to override.

(Or maybe we could keep just `AbstractTokenViewCall` but add methods,
```
protected boolean revertsOnMissingToken() {
    return true;
}

protected FullResult viewCallResultWith(ResponseCodeEnum status) {
    throw new UnsupportedOperationException();
}
```

?